library/src/blas2/rocblas_tpmv.cpp

```diff
@@ -0,0 +1,170 @@
+    template <>
+    constexpr char rocblas_tpmv_name<float>[] = "rocblas_stpmv";
+    template <>
+    constexpr char rocblas_tpmv_name<double>[] = "rocblas_dtpmv";
+    template <>
+    constexpr char rocblas_tpmv_name<rocblas_float_complex>[] = "rocblas_ctpmv";
+    template <>
+    constexpr char rocblas_tpmv_name<rocblas_double_complex>[] = "rocblas_ztpmv";
+
+    template <typename T>
+    rocblas_status rocblas_tpmv_impl(rocblas_handle    handle,
+                                     rocblas_fill      uplo,
+                                     rocblas_operation transA,
+                                     rocblas_diagonal  diag,
+                                     rocblas_int       m,
+                                     const T*          A,
+                                     T*                x,
+                                     rocblas_int       incx)
+    {
+        if(!handle)
+        {
+            return rocblas_status_invalid_handle;
+        }
+
+        auto layer_mode = handle->layer_mode;
+        if(layer_mode
+           & (rocblas_layer_mode_log_trace | rocblas_layer_mode_log_bench
+              | rocblas_layer_mode_log_profile))
+        {
+            auto uplo_letter   = rocblas_fill_letter(uplo);
+            auto transA_letter = rocblas_transpose_letter(transA);
+            auto diag_letter   = rocblas_diag_letter(diag);
+            if(layer_mode & rocblas_layer_mode_log_trace)
+            {
+                log_trace(handle, rocblas_tpmv_name<T>, uplo, transA, diag, m, A, x, incx);
+            }
+
+            if(layer_mode & rocblas_layer_mode_log_bench)
+            {
+                log_bench(handle,
+                          "./rocblas-bench",
+                          "-f",
+                          "tpmv",
+                          "-r",
+                          rocblas_precision_string<T>,
+                          "--uplo",
+                          uplo_letter,
+                          "--transposeA",
+                          transA_letter,
+                          "--diag",
+                          diag_letter,
+                          "-m",
+                          m,
+                          "--incx",
+                          incx);
+            }
+
+            if(layer_mode & rocblas_layer_mode_log_profile)
+            {
+                log_profile(handle,
+                            rocblas_tpmv_name<T>,
+                            "uplo",
+                            uplo_letter,
+                            "transA",
+                            transA_letter,
+                            "diag",
+                            diag_letter,
+                            "M",
+                            m,
+                            "incx",
+                            incx);
+            }
+        }
+
+        if(uplo != rocblas_fill_lower && uplo != rocblas_fill_upper)
+        {
+            return rocblas_status_not_implemented;
+        }
```

Comment:
There is now rocblas_status_invalid_value which I think is better as we don't expect to ever implement support other cases.

Reply:
Good point

Reply:
done.